streams/src/main/java/org/apache/kafka/streams/StoreQueryParams.java

```diff
@@ -0,0 +1,138 @@
+
+import org.apache.kafka.streams.state.QueryableStoreType;
+
+import java.util.Objects;
+
+/**
+ * Represents all the query options that a user can provide to state what kind of stores it is expecting.
+ * The options would be whether a user would want to enable/disable stale stores
+ * or whether it knows the list of partitions that it specifically wants to fetch.
+ * If this information is not provided the default behavior is to fetch the stores for all the partitions
+ * available on that instance for that particular store name.
+ * It contains a partition, which for a point queries can be populated from the {@link KeyQueryMetadata}.
+ */
+public class StoreQueryParams<T> {
+
+    private Integer partition;
+    private boolean staleStores;
+    private final String storeName;
+    private final QueryableStoreType<T> queryableStoreType;
+
+    private StoreQueryParams(final String storeName, final QueryableStoreType<T>  queryableStoreType) {
+        this.storeName = storeName;
+        this.queryableStoreType = queryableStoreType;
+    }
+
+    public static <T> StoreQueryParams<T> fromNameAndType(final String storeName,
+                                                          final QueryableStoreType<T>  queryableStoreType) {
+        return new StoreQueryParams<T>(storeName, queryableStoreType);
+    }
+
+    /**
+     * Set a specific partition that should be queried exclusively.
+     *
+     * @param partition   The specific integer partition to be fetched from the stores list by using {@link StoreQueryParams}.
+     *
+     * @return String storeName
+     */
+    public StoreQueryParams<T> withPartition(final Integer partition) {
+        final StoreQueryParams<T> storeQueryParams = StoreQueryParams.fromNameAndType(this.storeName(), this.queryableStoreType());
+        storeQueryParams.partition = partition;
+        storeQueryParams.staleStores = this.staleStores;
+        return storeQueryParams;
+    }
+
+    /**
+     * Enable querying of stale state stores, i.e., allow to query active tasks during restore as well as standby tasks.
+     *
+     * @return String storeName
+     */
+    public StoreQueryParams<T> enableStaleStores() {
+        final StoreQueryParams<T> storeQueryParams = StoreQueryParams.fromNameAndType(this.storeName(), this.queryableStoreType());
+        storeQueryParams.partition = this.partition;
+        storeQueryParams.staleStores = true;
+        return storeQueryParams;
+    }
+
+    /**
+     * Get the store name for which key is queried by the user.
+     *
+     * @return String storeName
+     */
+    public String storeName() {
+        return storeName;
+    }
+
+    /**
+     * Get the queryable store type for which key is queried by the user.
+     *
+     * @return QueryableStoreType queryableStoreType
+     */
+    public QueryableStoreType<T> queryableStoreType() {
+        return queryableStoreType;
+    }
+
+    /**
+     * Get the partition to be used to fetch list of stores.
```

Comment:
`to fetch list of stores` -- that describe some internal implementation detail.
Better: `Get the store partition that will be queried.`

Reply:
Done.